src/components/tx/modals/TokenTransferModal/ReviewSpendingLimitTx.tsx

```diff
@@ -0,0 +1,134 @@
+import { parseUnits } from '@ethersproject/units'
+import { EMPTY_DATA, ZERO_ADDRESS } from '@gnosis.pm/safe-core-sdk/dist/src/utils/constants'
+import useSafeInfo from '@/hooks/useSafeInfo'
+import { Errors, logError } from '@/services/exceptions'
+import ErrorMessage from '@/components/tx/ErrorMessage'
+import { useCurrentChain } from '@/hooks/useChains'
+import { dispatchSpendingLimitTxExecution } from '@/services/tx/txSender'
+import { getTxOptions } from '@/utils/transactions'
+
+export type SpendingLimitTxParams = {
+  safeAddress: string
+  token: string
+  to: string
+  amount: BigNumberish
+  paymentToken: string
+  payment: BigNumberish
+  delegate: string
+  signature: BytesLike
+}
+
+const ReviewSpendingLimitTx = ({ params, onSubmit }: ReviewTokenTxProps): ReactElement => {
+  const [isSubmittable, setIsSubmittable] = useState<boolean>(true)
+  const [submitError, setSubmitError] = useState<Error | undefined>()
+  const chainId = useChainId()
+  const currentChain = useCurrentChain()
+  const provider = useWeb3()
+  const { safeAddress } = useSafeInfo()
+  const { balances } = useBalances()
+  const token = balances.items.find((item) => item.tokenInfo.address === params.tokenAddress)
+  const spendingLimit = useSpendingLimit(token?.tokenInfo)
+
+  const txParams: SpendingLimitTxParams = useMemo(
+    () => ({
+      safeAddress,
+      token: spendingLimit?.token || ZERO_ADDRESS,
+      to: params.recipient,
+      amount: parseUnits(params.amount, token?.tokenInfo.decimals).toString(),
+      paymentToken: ZERO_ADDRESS,
+      payment: 0,
+      delegate: spendingLimit?.beneficiary || ZERO_ADDRESS,
+      signature: EMPTY_DATA,
+    }),
+    [
+      params.amount,
+      params.recipient,
+      safeAddress,
+      spendingLimit?.beneficiary,
+      spendingLimit?.token,
+      token?.tokenInfo.decimals,
+    ],
+  )
+
+  const { gasLimit, gasLimitLoading } = useSpendingLimitGas(txParams)
+
+  const [advancedParams, setManualParams] = useAdvancedParams({ gasLimit })
+
+  const handleSubmit = async (e: SyntheticEvent) => {
+    e.preventDefault()
+    if (!provider) return
+
+    setIsSubmittable(false)
+    setSubmitError(undefined)
+
+    const txOptions = getTxOptions(advancedParams, currentChain)
+
+    try {
+      await dispatchSpendingLimitTxExecution(txParams, txOptions, chainId, provider)
+
+      onSubmit(null)
+    } catch (err) {
+      logError(Errors._801, (err as Error).message)
+      setIsSubmittable(true)
+      setSubmitError(err as Error)
+    }
+  }
+
+  const submitDisabled = !isSubmittable || gasLimitLoading
+
+  return (
+    <form onSubmit={handleSubmit}>
+      <DialogContent>
+        <Typography variant="body2" mb={4}>
+          Spending limit transactions only appear in the interface once they are successfully mined and indexed. Pending
+          transactions can only be viewed in your signer wallet application or under your owner wallet address through a
+          Blockchain Explorer.
```

Comment:
I don't follow this. I think it could be rephrased a bit to be clearer. Might be best if we run through it together.

Reply:
I just copied it from safe-react

Reply:
Alright, rereading it made a bit more sense. I'd tweak it slightly though.
```suggestion
          transactions can only be viewed in your signer wallet application or under your wallet address on a
          Blockchain Explorer.
```